content/adding-login-to-your-nextjs-app-using-the-app-directory-and-supertokens/index.md

```diff
@@ -0,0 +1,815 @@
+
+In our APIs we will use `ensureSuperTokensInit` to make sure SuperTokens is initialised before we use any functionality from the SDK.
+
+Also modify the `dev` script in the `package.json` to:
+
+```json
+{
+    "scripts": {
+        "dev": "next dev -p 3000",
+    },
+}
+```
+
+We do this because in this example we want the project to run on port `3000`, if you want to keep the default port or use another port you can modify this script. Keep in mind that if you use a different port, you also need to modify the `apiDomain` and `websiteDomain` values inside the `/app/config/backend.ts` file.
+
+### Adding SuperTokens' API routes
+
+The SuperTokens backend SDKs expose a set of API routes that we can use in our project, this makes it easier to add login to our app because we don't have to build this logic ourselves.
+
+Start by creating a new file `/app/api/auth/[...path]/route.ts`. Adding `[...path]` as a folder makes sure that the `route.ts` file is called for all network calls made to the `/auth/*` paths. Add the following content to the `route.ts` file:
+
+```ts
+import { getAppDirRequestHandler } from 'supertokens-node/nextjs';
+import { NextRequest, NextResponse } from 'next/server';
+import { ensureSuperTokensInit } from '../../../config/backend';
+
+ensureSuperTokensInit();
+
+const handleCall = getAppDirRequestHandler(NextResponse);
+
+export async function GET(request: NextRequest) {
+  const res = await handleCall(request);
+  if (!res.headers.has('Cache-Control')) {
+    // This is needed for production deployments with Vercel
+    res.headers.set(
+      'Cache-Control',
+      'no-cache, no-store, max-age=0, must-revalidate'
+    )
+  }
+  return res;
+}
+
+export async function POST(request: NextRequest) {
+  return handleCall(request);
+}
+
+export async function DELETE(request: NextRequest) {
+  return handleCall(request);
+}
+
+export async function PUT(request: NextRequest) {
+  return handleCall(request);
+}
+
+export async function PATCH(request: NextRequest) {
+  return handleCall(request);
+}
+
+export async function HEAD(request: NextRequest) {
+  return handleCall(request);
+}
+```
+
+We call `ensureSuperTokensInit` to make sure SuperTokens is always initialised when any of the SuperTokens API routes are called. `getAppDirRequestHandler` is a helper function that the SuperTokens SDK provides, this adds request parsing, error handling etc so that we dont have to manually check for individual API results.
+
+And thats it, the SuperTokens backend SDK is setup. To verify that we have set it up correctly you can open http://localhost:3000/api/auth/signup/email/exists?email=somerandomemail@email.com in your browser, if it is all working correctly you should see a response similar to:
+
+```json
+{"status":"OK","exists":false}
+```
+
+You can see the [full API spec](https://supertokens.com/docs/fdi) to know all the API routes that the SuperTokens backend SDKs expose.
+
+### Adding SuperTokens to the frontend
```

Comment:
Adding "Steps" to the headers might make this more consumable since there are both frontend and backend setup steps.